.github/workflows/go.yml

```diff
@@ -0,0 +1,50 @@
+name: Golang CI
+
+'on':
+  - push
+  - pull_request
+
+jobs:
+  verify-goimports:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+          stable: true
+
+      - name: Install goimports
+        run: go get golang.org/x/tools/cmd/goimports
+
+      - name: Check code
+        uses: actions/checkout@v2
+      - run: goimports -w .
+
+      - name: Verify no changes from goimports
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            git status    # These are the files that failed to pass the check.
+            exit 1
+          fi
+
+  verify-gomod:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+          stable: true
+
+      - name: Check code
+        uses: actions/checkout@v2
+
+      - run: find . -name go.mod -execdir go mod tidy \;
```

Comment:
would it make sense to codify some of this as top-level make rules so we can run them locally?

Reply:
That's a good idea - I think we can do that as a separate PR though!